file.go

```diff
@@ -455,17 +461,35 @@ type filePages struct {
 
 	section *io.SectionReader
 	rbuf    *bufio.Reader
+	limit   io.LimitedReader
 
 	// This buffer holds compressed pages in memory when they are read; we need
 	// to read whole pages because we have to compute the checksum prior to
 	// exposing the page to the application.
-	compressedPageData []byte
+	compressedPageData *bytes.Buffer
 
 	page filePage
 	skip int64
 }
 
+func (r *filePages) Close() error {
```

Comment:
almost wonder about using Reset() as the name here if there's nothing that could return an error

Reply:
I guess just a general comment is I'd expect Close() to actually be associated with some resource that is "open" like a file handle or database connection or something.

It seems we're using it a lot of the time to reclaim buffer space.

As someone who's not super familiar, it makes it pretty difficult to reason about, what is actually happening when I call Close() on a given resource.

Reply:
I've added `Close` methods on types that act as streams/readers, for example in this case the `parquet.Pages` interface is defined as:
```go
type Pages interface {
    PageReader
    io.Closer
}

type PageReader interface {
    ReadPage() (Page, error)
}
```
I'm under the impression that this is a common pattern, it reminds of `io.ReadCloser` for example. Closing the reader helps release resources, but it also acts as a mechanism to interrupt the stream, where after a call to `Close` the reads will return `io.EOF`.

What do you think?